model/variable.cc

```diff
@@ -453,10 +462,16 @@ void VariableBase::draw(cairo_t *cairo) const
   rv.angle=angle+(notflipped? 0: M_PI);
 
   // parameters of icon in userspace (unscaled) coordinates
-  float w, h, hoffs;
+  float w, h, hoffs, fontFactor;
   w=rv.width()*z; 
   h=rv.height()*z;
+  fontFactor=min(iWidth()/rv.width(),iHeight()/rv.height());
```

Comment:
fontFactor=min(1.0, min(iWidth()/rv.width(),iHeight()/rv.height()));

would eliminate the conditional on line 472. Just need to call setFontSize()